baseplate/lib/message_queue.py

```diff
@@ -103,22 +134,72 @@ def put(self, message: bytes, timeout: Optional[float] = None) -> None:
         raise TimedOutError
 
     def unlink(self) -> None:
-        """Remove the queue from the system.
+        self.queue.unlink()
 
-        The queue will not leave until the last active user closes it.
+    def close(self) -> None:
+        self.queue.close()
 
-        """
-        self.queue.unlink()
+
+class InMemoryMessageQueue(MessageQueue):
+    """An in-memory inter process message queue."""  # Used by the sidecar
+
+    def __init__(self, name: str, max_messages: int):
+        self.queue: q.Queue = q.Queue(max_messages)
+        self.max_messages = max_messages
+        self.name = name
+
+    def get(self, timeout: Optional[float] = None) -> bytes:
+        try:
+            message = self.queue.get(timeout=timeout)
+            self.queue.task_done()
+            return message
+        except q.Empty:
+            raise TimedOutError
+
+    def put(self, message: bytes, timeout: Optional[float] = None) -> None:
+        try:
+            self.queue.put(message, timeout=timeout)
+        except q.Full:
+            raise TimedOutError
 
     def close(self) -> None:
-        """Close the queue, freeing related resources.
+        """Not implemented for in-memory queue"""
 
-        This must be called explicitly if queues are created/destroyed on the
-        fly. It is not automatically called when the object is reclaimed by
-        Python.
 
-        """
-        self.queue.close()
+class RemoteMessageQueue(MessageQueue):
+    def __init__(self, name: str, max_messages: int, host: str = "127.0.0.1", port: int = 9090):
+        self.queue_name = name
+        self.max_messages = max_messages
+        self.host = host
+        self.port = port
+
+    def connect(self):
+        # Establish a connection with the queue server
+        transport = TSocket.TSocket(self.host, self.port)
+        self.transport = TTransport.TBufferedTransport(transport)
+        protocol = TBinaryProtocol.TBinaryProtocol(self.transport)
+        self.client = RemoteMessageQueueService.Client(protocol)
+        self.transport.open()
+
+    def get(self, timeout: Optional[float] = None) -> bytes:
+        # Call the remote server and get an element for the correct queue
+        try:
+            self.connect()
+            return self.client.get(self.queue_name, self.max_messages, timeout).value
+        except ThriftTimedOutError:
+            raise TimedOutError
+
+    def put(self, message: bytes, timeout: Optional[float] = None) -> None:
+        # Call the remote server and put an element on the correct queue
+        # Will create the queue if it doesnt exist
+        try:
+            self.connect()
+            self.client.put(self.queue_name, self.max_messages, message, timeout)
+        except ThriftTimedOutError:
```

Comment:
This is confusing, if you look at the import on L17 `ThriftTimedOutError` is a `TimedOutError`. So why are you catching it and rerasing the same error type but empty of any info?

Reply:
There are two `TimedOutError`s - one defined in `message_queue.py` and one defined in the Thrift types. I'll include the info from the original exception, though

Reply:
Clarifying the naming to differentiate them - now there is a `TimedOutError` and a `ThriftTimedOutError` - hopefully that's clearer